test/highlight/records.gleam

```diff
@@ -0,0 +1,13 @@
+pub type User {
+  User(name: String)
+  // <- type
+  //    ^ property
+  //          ^ type
+}
+
+pub fn new(name: String) {
+  //        ^ variable.parameter
+  User(name: name)
+  //    ^ property
```

Comment:
this highlight was mistakenly being overridden to `variable.parameter` which looked quite confusing in the highlights